src/util/translation.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_TRANSLATION_H
+#define BITCOIN_UTIL_TRANSLATION_H
+
+#include <tinyformat.h>
+
+#include <utility>
+
+/**
+ * Bilingual messages:
+ *   - in GUI: user's native language + untranslated (i.e. English)
+ *   - in log and stderr: untranslated only
+ */
+struct bilingual_str {
+    std::string original;
+    std::string translated;
+};
+
+namespace tinyformat {
+template <typename... Args>
+bilingual_str format(const bilingual_str& fmt, const Args&... args)
+{
+    return bilingual_str{format(fmt.original, args...), format(fmt.translated, args...)};
+}
+} // namespace tinyformat
+
+/** Translate a message to the native language of the user. */
+const extern std::function<std::string(const char*)> G_TRANSLATION_FUN;
+
+/**
+ * Translation function.
+ * If no translation function is set, simply return the input.
+ */
+inline std::string _(const char* psz)
```

Comment:
What about this?

```suggestion
inline bilingual_str _(const char* psz)
```

Have you seen my comment here: https://github.com/bitcoin/bitcoin/pull/16224#issuecomment-502737888?

Reply:
> Have you seen my comment here: #16224 (comment)?

Sure!

IIUC, your suggestion implies some kind of type conversion (from `bilingual_str` to `std::string`) for arguments of some functions and in assignment statements?

Reply:
If you don't need the original string, it can be dropped with `_().translated`, no? I understand that the changes are more disruptive, but it seems clearer than the unused optional argument.

It could probably be done as a scripted-diff.